groups:
- name: external-dns.rules
  rules:
  - alert: ExternalDNSRegistryErros
    expr: sum(rate(external_dns_registry_errors_total[1m])) by (env, region) > 0
    for: 5m
    labels:
      severity: s4
      alert_type: cause
    annotations:
      description: |
        ExternalDNS is getting errors on {{.labels.region}} when communicating
        with GCP. Some possible reasons are network errors, GCP degradation,
        missing or wrong credentials. Check the ExternalDNS logs for more
        information.
      runbook: docs/external-dns/README.md
      title: External DNS Registry errors
  - alert: ExternalDNSSourceErros
    expr: sum(rate(external_dns_source_errors_total[1m])) by (env, region) > 0
    for: 5m
    labels:
      severity: s4
      alert_type: cause
    annotations:
      description: |
        ExternalDNS is getting source errors on {{.labels.region}}.  Some
        possible reasons are connection errors to the Kubernetes API or missing
        RBAC permissions. It can also stem from incompatible configuration in
        the objects itself like invalid characters, processing a broken
        fqdnTemplate, etc. Check the ExternalDNS logs for more information.
      runbook: docs/external-dns/README.md
      title: External DNS Source errors
  - alert: ExternalDNSStale
    expr: time() - external_dns_controller_last_sync_timestamp_seconds > 60
    for: 5m
    labels:
      pager: pagerduty
      severity: s1
      alert_type: symptom
    annotations:
      description: |
        ExternalDNS has not synced successfully with Google Cloud DNS on
        {{.labels.region}} in the last five minutes. This can lead to missing or
        outdated DNS entries, which might lead to service disruption. Check the
        ExternalDNS logs for more information.
      runbook: docs/external-dns/README.md
      title: External DNS is stale
